### DISCUSSION TITLE
fastfetch with custom ascii art

### DISCUSSION THREAD

**Discussion by imgustavo** [fastfetch-cli/fastfetch#1191] [Q&A] [answered]
Hello. If anybody can help me out with this: I tried to change fastfetch with my custom ascii logo. I did this before with neofetch. But now its not working for some reason. I've already check many tutorials and here the github's project and did what it says. Maybe am i missing something? This is how i edited my config.jsonc:
(note that i commented the first 2 source paths, i tried to make it work like that, but it didnt... so i left it like this: commented)

```
 {
  "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
   "logo": {
          //"source": "~/.config/fastfetch/logo.txt"
          //"source": "~/.config/fastfetch/logos/skull.txt"
           },
  "modules": [
    "title",
    "separator",
    "os",
    "host",
    "kernel",
    "packages",
    "shell",
    "de",
    "wm",
    "wmtheme",
    "theme",
    "icons",
    "font",
    "terminal",
    "terminalfont",
    "cpu",
    "memory",
    "localip",
    "publicip",
    "break",
    "colors"
  ]
}
```

**Comment by CarterLi** [comment]
You should check this: https://github.com/fastfetch-cli/fastfetch/wiki/Migrate-Neofetch-Logo-To-Fastfetch

If it still doesn't work, you should upload your ascii logo file

**Reply by imgustavo** [reply]
> You should check this: https://github.com/fastfetch-cli/fastfetch/wiki/Migrate-Neofetch-Logo-To-Fastfetch
> 
> If it still doesn't work, you should upload your ascii logo file

I'm really lost. 
1- I copied and pasted the ascii art as in the example given in your link "Migrate Neofetch Logo To Fastfetch", modified to work in fastfetch with the name "test.txt"
2- I copied and pasted in this path:  /.config/fastfetch/test.txt, so the result with be the command:
fastfetch -l /.config/fastfetch/test.txt --logo-color-1 white --logo-color-2 red --logo-color-3 '38;2;23;147;209'
3- The ascii art that i get is the default one for linux mint (my distro)

I really dont get what im doing wrong. As i mentioned i made this work before in neofetch and wasn't hard for me

**Reply by CarterLi** [reply]
Shouldn't it be `~/.config/fastfetch/test.txt`?

**Reply by CarterLi** [reply]
You can always use `--show-errors` to see what's going wrong.

**Comment by imgustavo** [comment, marked as answer]
I have found a workaround for this:

- I downloaded the source code of fastfetch
git clone https://github.com/fastfetch-cli/fastfetch.git
cd fastfetch

- Join in the folder fastfetch -> src -> logo -> ascii -> and edited  with my custom ascii one of the default ascii that fastfetch already comes with

- Then i build from source code fastfetch following the instructions from here: https://github.com/fastfetch-cli/fastfetch/wiki/Building
(I have LMDE6: so to build from source code i had to install cmake, no problem is in the software manager)

- Then in console:
make
sudo make install

Problem solved.

Notes aside: this wont overwrite your config.jsonc with your previous fastfetch installation, but still you can make a backup for it just in case. In my case i did it, but it didnt need it.

**Reply by wolfa4** [reply]
Thank you for that explanation. I just followed what you said and was able to get it running. The only issue I ran into was getting an error: 

```
-- The CXX compiler identification is unknown

CMake Error at CMakeLists.txt:146 (enable_language):

  No CMAKE_CXX_COMPILER could be found.
```

which I solved by installing the following packages:

`sudo dnf install make automake gcc gcc-c++`

I just went from 20 years of windows ->linux in DEC'24 -> custom build of fastfetch with custom ascii.

**Comment by S-e-r-a-p-h-i-n-e** [comment]
I think I actually see the problem here. You forgot to add ```"type": "file"``` under your source.

If you're still interested, here's your fixed, config.jsonc (Remove the comment tag on your preferred logo)
```
 {
  "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
   "logo": {
          //"source": "~/.config/fastfetch/logo.txt",
          //"source": ~/.config/fastfetch/logos/skull.txt",
          "type": "file"
           },
  "modules": [
    "title",
    "separator",
    "os",
    "host",
    "kernel",
    "packages",
    "shell",
    "de",
    "wm",
    "wmtheme",
    "theme",
    "icons",
    "font",
    "terminal",
    "terminalfont",
    "cpu",
    "memory",
    "localip",
    "publicip",
    "break",
    "colors"
  ]
}
```

Let me know if you encounter any problems.

**Reply by morfoloo** [reply]
> I think I actually see the problem here. You forgot to add `"type": "file"` under your source.

Hello. I have an error with fastfetch: JsonConfig Error: Unknown logo key 
Here is my config:
{
  "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
  "logo": {
    "source": "home/morfoloo/.local/share/fastfetch/ascii/arch.txt",
    "type": "file",
    "paddingLeft": 2,
    "paddingTop": 1
  },
  "display": {
    "separator": " : "
  },
  "modules": [
    {
      "type": "os",
      "key": " OS",
      "format": "{2}",
      "keyColor": "red"
    },
    {
      "type": "kernel",
      "key": " Kernel",
      "format": "{2}",
      "keyColor": "red"
    },
    {
      "type": "cpu",
      "key": " CPU",
      "format": "{1} @ {7}",
      "keyColor": "blue"
    },
    {
      "type": "memory",
      "key": " Mem",
      "keyColor": "magenta"
    },
    {
      "type": "wm",
      "key": "󱗃 WM",
      "format": "{2}",
      "keyColor": "yellow"
    },
    {
      "type": "terminal",
      "key": " Term",
      "keyColor": "yellow"
    },
    {
      "type": "uptime",
      "key": "󱫐 Uptime",
      "keyColor": "green"
    },
    {
      "type": "colors",
      "symbol": "circle"
    }
  ]
}

**Comment by froux07** [comment]
For padding :
     	"padding": {
             "top": 6,
             "left": 1,
             "right": 1
        }
